WordPress/gutenberg
# Adapt site editor layout widths based on viewport size

https://github.com/WordPress/gutenberg/assets/846565/c21d3157-97e2-4243-8bf8-7d1bfdbaed75



## What?
Update width of site editor layout areas based on viewport size. 

## Why?
The existing one-size-fits-all dimensions do not scale elegantly, and the navigation sidebar is arguably too wide, especially at smaller screen sizes.

* The dark navigation sidebar width scales from 260px to 308px.
* The content area width scales from 360px to 488px.

## Testing Instructions
1. Check out this branch
2. `npm run build`
3. Open the site editor
4. Notice that the width of the dark navigation sidebar, and the content area (in data views list layout) width changes with the viewport.

<details><summary>Original PR</summary>## What?
Reduce the width of the navigation sidebar in the site editor from 360px to 320px.

## Why?
Currently the navigation sidebar and the list layout content frame are roughly equal in width (360px and 380px respectively). Given the nature of the content in each element I'm not sure this is appropriate. 

As the Navigation sidebar consists mostly of menu items it's width can be reduced slightly. This attempts to install more appropriate visual balance between sidebar / content, and provides more space for page contents / preview frames on the right.

The value isn't necessary a proposal, there may be scope to go narrower, or to adjust the width of the content frame instead. The PR mostly aims to start the conversation.

I acknowledge it may be preferable to wait until we have a more concrete vision for the broader admin design for making this change.  

## Testing Instructions
1. Check out this branch
2. `npm run build`
3. Open the site editor
4. Navigate around observing the narrower navigation sidebar

### Before
<img width="908" alt="Screenshot 2024-01-30 at 13 22 45" src="https://github.com/WordPress/gutenberg/assets/846565/bf77e81b-332f-4e38-93a0-560751ec65c0">


### After
<img width="871" alt="Screenshot 2024-01-30 at 13 20 41" src="https://github.com/WordPress/gutenberg/assets/846565/cd4b3a5d-730d-49fe-b2c6-4354f986d846">
</details>
